lean4-mode/lean4-mode.el

```diff
@@ -208,9 +208,30 @@ Invokes `lean4-mode-hook'.
 ;;;###autoload
 (push '("\\.lean$" . lean4-mode) auto-mode-alist)
 
+(defun lean--version ()
+  (let ((version-line (car (process-lines "lean" "-v"))))
```

Comment:
Please use `lean4-get-executable`. Also, have you tested this when elan first has to download Lean and prints its own output first?

Reply:
I hadn't thought of that. Is there a way of asking if elan if the required lean version needs to be downloaded?

Reply:
I think it's sufficient to just parse the last line instead